realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -499,25 +519,17 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                 } else {
                     // update the instance with the new pointer
                     realmResults.swapTableViewPointer(query.getValue());
-
-                    // it's dangerous to notify the callback about new results before updating
-                    // the pointers, because the callback may use another RealmResults not updated yet
-                    // this is why we defer the notification until we're done updating all pointers
-                    callbacksToNotify.add(realmResults);
+                    realmResults.syncIfNeeded();
+                    resultsToBeNotified.add(realmResults);
```

Comment:
I will double check this. It feels it will be added to the list in the following `collectSyncRealmResultsCallbacks(resultsToBeNotified);`